low-level/files/opt/seagate/sspl/setup/resource_map/storage.py

```diff
@@ -44,8 +44,14 @@ def __init__(self):
         super().__init__()
         self.validate_storage_type_support()
         self.storage_frus = {
-            "controllers": self.get_controllers_info
-            }
+            "controllers": self.get_controllers_info,
+            "logical_volumes": self.get_logical_volumes_info,
+            "disk_groups": self.get_disk_groups_info
+        }
+        # Note: Please make sure logical volumes get executed 1st before the disk.
```

Comment:
Logic need to take care of this code invocation by itself if logical volume info not available, given health consumer can ask data specifically for disk-groups, without any logical volumes info necessarily.

Reply:
Yes, That is already handled on line no. 183.

Reply:
> if pool_sr_no in self.dg_vol_map:
Thats line 183
Else case is not to fill in volume details for that disk group
dg_vol_map[] is filled up if someone calls logical volume health collection otherwise disk group wont have vome mappings, ideally disk group just retrieve volume details & then fill in.

Reply:
Done. added mapping logic in the disk group itself. now it is not dependant on logical volume function.